### PR TITLE
Add peg 0.1.18

### PR DESCRIPTION
**Pull request by fatliverfreddy**
Specify library name and version:  **peg/0.1.18**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/peg/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def build(self):
+        if self.settings.os == "Windows":
+            with tools.chdir(self._source_subfolder):
+                with tools.vcvars(self.settings):
+                    msbuild = MSBuild(self)
+                    if self.settings.arch == "x86":
+                        msbuild.build_env.link_flags.append("/MACHINE:X86")
+                    elif self.settings.arch == "x86_64":
+                        msbuild.build_env.link_flags.append("/SAFESEH:NO /MACHINE:X64")
```
Comment (prince-chrismc) [inline recipes/peg/all/conanfile.py:37]: I would have expected the build helper to take case of these 🤔

File: recipes/peg/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    msbuild.build(project_file="peg.sln", platforms={"x86": "Win32"}, build_type="Release", force_vcvars=True, arch="x86")
```
Comment (prince-chrismc) [inline recipes/peg/all/conanfile.py:38]: https://docs.conan.io/en/latest/reference/build_helpers/visual_studio.html

> To inject the flags corresponding to the compiler.runtime, build_type and compiler.cppstd settings, this build helper also generates a properties file (in the build folder) that is passed to :command:MSBuild with :command:/p:ForceImportBeforeCppTargets="conan_build.props".

This code seems to completely ignore the build settings 😕

File: recipes/peg/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import os
+
+class PegConan(ConanFile):
+    name = "peg"
+    license = "MIT"
+    description = "peg and leg are tools for generating recursive-descent parsers"
+    topics = ("conan", "peg", "leg", "parser")
+    homepage = "https://www.piumarta.com/software/peg/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (prince-chrismc) [inline recipes/peg/all/conanfile.py:11]: I can not remember what the rules are for tools... https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-installer-packages-remove-some-settings-from-their-package-id

Please follow the docs for modifying the `package_id`

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

File: recipes/peg/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from conans import ConanFile
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+
+    def test(self):
+        if not self.settings.os == "Windows":
```
Comment (memsharded) [inline recipes/peg/all/test_package/conanfile.py:7]: Why would the application not run in Windows? This test should be running also the Windows exe.

```suggestion
        if self.settings.os != "Windows":
```

File: recipes/peg/all/conanfile.py (hunk 2 above)
Comment (memsharded) [reply recipes/peg/all/conanfile.py:38]: Agree. @fatliverfreddy if this package would not build with other configurations, then that should be raised as ``ConanInvalidConfiguration`` in the ``validate()`` method. You can see a few cases in the repo: https://github.com/conan-io/conan-center-index/search?q=validate%28self%29&type=code

But it is not necessary to hardcode the settings, the build helper will take care of that.

**Review by memsharded — COMMENTED** [review]
Hi @fatliverfreddy 

Could you please check the open issues? With just a few changes, this could be good to be merged. Thanks!

**Pull request closed without merging**